lib/cli/interactive-setup/aws-credentials.js

```diff
@@ -86,33 +178,141 @@ const steps = {
+          name: 'skipProviderSetup',
+        });
+
+        inquirerPrompt.then(() => resolve(null));
+      }, timeoutDuration);
+
+      onEvent = (event) => {
+        if (inquirerPrompt) {
+          // Disable inquirer prompt asking to skip without setting provider
+          inquirerPrompt.ui.close();
+        }
+
+        clearTimeout(showSkipPromptTimeout);
+        resolve(event);
+      };
+    });
+
+    // Listen for `provider.created` event to detect creation of new provider
+    const sdk = await getSdkInstance(orgName);
+    await sdk.connect({
+      orgName,
+      onEvent,
+      filter: {
+        events: ['provider.created'],
+      },
+    });
+
+    let maybeEvent;
+    try {
+      maybeEvent = await p;
+    } finally {
+      sdk.disconnect();
+    }
+
+    if (maybeEvent) {
+      process.stdout.write(`\nDetected creation of provider: "${maybeEvent.data.object.alias}"\n`);
+    } else {
+      // TODO: ADD MESSAGING THAT YOU SKIPPED SETUP OF THE PROVIDER
+    }
+
+    // TODO: CONSIDER IF WE SHOULD CREATE DIRECT LINK for default provider
+  },
+  // TODO: TESTS
+  linkProviderToService: async (app, service, orgName, providerUid) => {
+    const sdk = await getSdkInstance(orgName);
+    const linkType = 'service';
+    const linkUid = getProviderLinkUid(app, service);
+    const orgUid = await getOrgUidByName(orgName);
+    return sdk.createProviderLink(orgUid, linkType, linkUid, providerUid);
+  },
 };
 
 module.exports = {
-  async isApplicable({ configuration }) {
+  async isApplicable(context) {
+    const { configuration } = context;
     if (
       _.get(configuration, 'provider') !== 'aws' &&
       _.get(configuration, 'provider.name') !== 'aws'
     ) {
       return false;
     }
     if (new AWS.S3().config.credentials) return false;
-    return !(await awsCredentials.resolveFileProfiles()).size;
+    if ((await awsCredentials.resolveFileProfiles()).size) return false;
+
+    // TODO: TESTS
+    const orgName = _.get(configuration, 'org');
+    if (orgName) {
+      const providers = await getProviders(orgName);
+      const hasDefaultProvider = providers.some((provider) => provider.isDefault);
+
+      if (hasDefaultProvider) return false;
+    }
+
+    return true;
   },
   async run(data) {
-    process.stdout.write(
-      'No AWS credentials were found on your computer, ' +
-        'you need these to host your application.\n\n'
-    );
-
     return module.exports.runSteps(data);
   },
   steps,
   runSteps: async (context) => {
-    if (!(await steps.shouldSetupAwsCredentials())) return;
-    await steps.ensureAwsAccount();
+    const providers = await getProviders(context.configuration.org);
+    const credentialsSetupChoiceAnswer = await credentialsSetupChoice(providers);
+
+    if (credentialsSetupChoiceAnswer === CREDENTIALS_SETUP_CHOICE.CREATE_PROVIDER) {
+      try {
+        await steps.handleProviderCreation(context);
+      } catch (err) {
+        throw new ServerlessError(
```

Comment:
Try/catch scope feels as too broad here. I wouldn't introduce this wrap, but instead implement proper SDK communication error handling in context of `steps.handleProviderCreation` internals

Reply:
I went with this approach previously, but I'm wondering how granular/explicit we want to be with users e.g. should we show them errors like "Could not connect to Dashboard" when we try to grab a connection to the websocket? Not sure how much insight it provides

Reply:
I've adjusted the error handling, please let me know what do you think :+1: